Integration tests related to output console

Hema-Mathiyazhagan opened the pull request:
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

Issue: Adding integration tests related to output console. https://github.com/elyra-ai/elyra/issues/536
Testing: Tested the changes locally

karlaspuldaro reviewed (changes requested):
Hi Hema,
Thank you for addressing previous review comments in #2498
I'd like to add a few more points:
1. If you could please add a detailed description to the PR and link it to the issue it addresses  it'd be helpful to reviewers to understand the purpose of your PR.
2. Since you're using the Python editor for testing, please move the new tests up to Python Tests section to follow current file organization.
3. We want to avoid duplicated code, maybe since you need the open file step from `openFileAndCheckContent` function, I suggest it requires a few tweaks like:
- create a function for opening a file
- create function to check file content
- update existing calls to `openFileAndCheckContent`
- call `openFile` from new tests  - ok to group checking for output tab and buttons on the same test to save some time and steps
- check console output tab and buttons directly in the test body

karlaspuldaro commented on tests/integration/scripteditor.ts:
```diff
@@ -118,6 +118,9 @@ describe('Script Editor tests', () => {
     cy.get('button[title="Bottom"]').should('be.visible');
+    
+    // Close editor tab
+    cy.closeTab(-1);
```
Good catch, although I guess this will close the console output tab, maybe you'd need a second call to close the editor tab.

karlaspuldaro commented on tests/integration/scripteditor.ts:
```diff
@@ -109,6 +109,20 @@ describe('Script Editor tests', () => {
 
+  it('opens new output console', () => {
+    openFile('py');
+    cy.get('button[title="Run"]').click({ force: true });
```
I think you can remove the `{ force: true }` option since there should be only one element found at this point.

Hema-Mathiyazhagan commented:
> Hi Hema, Thank you for addressing previous review comments in #2498 I'd like to add a few more points:
> 
> 1. If you could please add a detailed description to the PR and link it to the issue it addresses  it'd be helpful to reviewers to understand the purpose of your PR.
> 2. Since you're using the Python editor for testing, please move the new tests up to Python Tests section to follow current file organization.
> 3. We want to avoid duplicated code, maybe since you need the open file step from `openFileAndCheckContent` function, I suggest it requires a few tweaks like:
> 
> * create a function for opening a file
> * create function to check file content
> * update existing calls to `openFileAndCheckContent`
> * call `openFile` from new tests  - ok to group checking for output tab and buttons on the same test to save some time and steps
> * check console output tab and buttons directly in the test body

Addressed this comments @karlaspuldaro

karlaspuldaro commented:
Looking good all green! 💚
Thank you for addressing the comments.
Just checking if there are more tests you're planning to add to this PR, otherwise I can approve it already.

Hema-Mathiyazhagan commented:
> Looking good all green! 💚 Thank you for addressing the comments. Just checking if there are more tests you're planning to add to this PR, otherwise I can approve it already.

@karlaspuldaro  The PR is ready for final review Karla.

karlaspuldaro commented on tests/integration/scripteditor.ts:
```diff
@@ -91,7 +92,7 @@ describe('Script Editor tests', () => {
 
     // Add some text to the editor
-    cy.get('span[role="presentation"]').type('print("test")');
+    cy.get('span[role="presentation"]').type('print("test")\n');
```
To clarify, this is being added here as suggested by @ajbozarth to overcome a new code completer element that sometimes blocks the toolbar buttons during testing runtime.

karlaspuldaro reviewed (approved):
This looks good! Thank you @Hema-Mathiyazhagan for contributing :)
